File: problems/C++/39_n_repeated_element_in_size_2n_array.cpp
//Problem Statement
//In a array A of size 2N, there are N+1 unique elements, and exactly one of
//these elements is repeated N times.
/* Return the element repeated N times.
Example :

Input: [1,2,3,3]
Output: 3 */
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int repeatedNTimes(vector<int>& A) {
        unordered_set<int> unique;
        for(auto val : A){
          if(unique.find(val)!=unique.end())
            return val;
          unique.insert(val);
        }
        return 0;
    }
};

int main(){
  vector<int> vec = {1,2,3,3};
  Solution sol;
  cout<<sol.repeatedNTimes(vec)<<endl;
  return 0;
}

File: problems/Kotlin/39_n_repeated_element_in_size_2n_array.kt
//Problem Statement
//In a array A of size 2N, there are N+1 unique elements, and exactly one of
//these elements is repeated N times.
/* Return the element repeated N times.
Example :

Input: [1,2,3,3]
Output: 3 */

class Solution {
    fun repeatedNTimes(A: IntArray): Int {
        var unique:HashSet<Int> = HashSet<Int>();

        for( value in A){
            if(unique.contains(value)){
                return value;
            }
            unique.add(value);
        }

        return 0;
    }
}

fun main(args:Array<String>){
  val vec:IntArray = intArrayOf(1,2,3,3);
  val sol:Solution = Solution();
  println(sol.repeatedNTimes(vec));
}
